src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -137,7 +137,7 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
 
         // In case we're in a module, try to resolve the relative path.
         if let Some(module_id) = parent_id.or(self.mod_ids.last().cloned()) {
-            let module_id = cx.tcx.hir().hir_id_to_node_id(module_id);
+            let module_id = cx.tcx.hir().local_def_id(module_id);
```

Comment:
Are you sure `module_id` can refer to a block here?
Does rustdoc test suite fail if `resolve_str_path_error` doesn't look into the `block_map` at all?

I wouldn't want to create definitions for all blocks, if they are used in rustdoc only.

Reply:
Good point! It seems to work.

Reply:
Great.